### PR TITLE
Modify custom_call_test for rocm 

### PR DESCRIPTION
**Pull request by sumesh13**

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/custom_call_test.cc [hunk 1]
```diff
@@ -13,28 +13,54 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
+#if GOOGLE_USE_CUDA
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/custom_call_test.cc]: @sumesh13 i am not sure if this is right

File: tensorflow/compiler/xla/service/gpu/custom_call_test.cc [hunk 2]
```diff
@@ -13,28 +13,54 @@ See the License for the specific language governing permissions and
...
 #include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/stream_executor/gpu/gpu_types.h"
+
+
+#if GOOGLE_CUDA
+#define gpuSuccess cudaSuccess
+#define gpuMemCpyAsync cudaMemcpyAsync
+#define gpuMemcpyDeviceToDevice cudaMemcpyDeviceToDevice
+#define gpuMemcpy cudaMemcpy
+#define  gpuMemcpyDeviceToHost cudaMemcpyDeviceToHost
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/custom_call_test.cc]: extra whitespace

File: tensorflow/compiler/xla/service/gpu/custom_call_test.cc [hunk 3]
```diff
@@ -13,28 +13,54 @@ See the License for the specific language governing permissions and
...
 #include "tensorflow/compiler/xla/status_macros.h"
 #include "tensorflow/compiler/xla/test_helpers.h"
 #include "tensorflow/compiler/xla/tests/client_library_test_base.h"
 #include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/stream_executor/gpu/gpu_types.h"
+
+
+#if GOOGLE_CUDA
+#define gpuSuccess cudaSuccess
+#define gpuMemCpyAsync cudaMemcpyAsync
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/custom_call_test.cc]: typo

File: tensorflow/compiler/xla/service/gpu/custom_call_test.cc [hunk 4]
```diff
@@ -13,28 +13,54 @@ See the License for the specific language governing permissions and
...
 #include "tensorflow/compiler/xla/service/custom_call_target_registry.h"
 #include "tensorflow/compiler/xla/status_macros.h"
 #include "tensorflow/compiler/xla/test_helpers.h"
 #include "tensorflow/compiler/xla/tests/client_library_test_base.h"
 #include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/stream_executor/gpu/gpu_types.h"
+
+
+#if GOOGLE_CUDA
+#define gpuSuccess cudaSuccess
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/custom_call_test.cc]: `gpuSuccess` is actually defined in `tensorflow/core/util/gpu_kernel_helper.h` already.

for the rest of the macros perhaps it's fine to leave them here as they are only used by this test case.

Comment (sumesh13) [reply tensorflow/compiler/xla/service/gpu/custom_call_test.cc]: tensorflow/core/util/gpu_kernel_helper.h  brings in quite a few symbols and dependencies. So not sure if its worth it....

Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/custom_call_test.cc]: got it. I'm fine leaving it here.

**Pull request merged**